### PR TITLE
[Draft] Add jruby to CI

### PR DESCRIPTION
**Pull request by tachyons**
Add JRuby and JRuby CI to expand the platform coverage.

### Manual Tests

Fixes https://github.com/Shopify/ruby-lsp/issues/687

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -35,7 +35,7 @@ jobs:
       fail-fast: false
       matrix:
         os: [ubuntu-latest, macos-latest, windows-latest]
-        ruby: ["3.0", "3.1", "3.2", "head"]
+        ruby: ["3.0", "3.1", "3.2", "head", "jruby", "jruby-head"]
```
Comment (vinistock) [inline .github/workflows/ci.yml:38]: I think just `jruby-head` is probably fine

**Review by vinistock — COMMENTED** [review]
Looks like CI is having platform issues. Not sure how you solve them exactly, is there a generic java platform that's not tied to a specific version?

File: .github/workflows/ci.yml (hunk 1 above)
Comment (tachyons) [reply .github/workflows/ci.yml]: @vinistock I tried that too, But it was still failing with cryptic sorbet error. I haven't worked with sorbet before to debug that.

**Comment by tachyons** [comment]
Screenshot 2023-12-19 at 7 40 08 AM
This is the error I am getting, Same issue is reported by another user [here](https://github.com/Shopify/vscode-ruby-lsp/issues/920)

**Comment by andyw8** [comment]
Thank you, it seems this related to the [patch](https://github.com/Shopify/ruby-lsp/blob/main/lib/core_ext/uri.rb) we have for `URI::Generic.from_path`.

As Sorbet [does not officially support](https://github.com/sorbet/sorbet/blob/master/docs/JRuby.md) JRuby, this may be a difficult one to solve...

**Pull request closed without merging**